x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +242,169 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
+		poolBalanceBefore := sdk.Coins{}
+		for _, asset := range test.msg.InitialLiquidity() {
+			poolBalanceBefore = poolBalanceBefore.Add(sdk.NewCoin(asset.Denom, gammKeeper.GetDenomLiquidity(suite.Ctx, asset.Denom)))
+		}
+
+		// createPool process
+		err := test.msg.Validate(suite.Ctx)
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		initialPoolLiquidity := test.msg.InitialLiquidity()
+
+		numAssets := initialPoolLiquidity.Len()
+		suite.Require().Equal(numAssets < swaproutertypes.MinPoolAssets, false)
+		suite.Require().Equal(numAssets > swaproutertypes.MaxPoolAssets, false)
+
+		fee := swaprouterKeeper.GetParams(suite.Ctx).PoolCreationFee
+		err = communityPoolKeeper.FundCommunityPool(suite.Ctx, fee, sender)
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		poolId := swaprouterKeeper.GetNextPoolId(suite.Ctx)
+		swaprouterKeeper.SetNextPoolId(suite.Ctx, poolId+1)
+
+		pool, err := test.msg.CreatePool(suite.Ctx, poolId)
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		swaprouterKeeper.SetPoolRoute(suite.Ctx, poolId, test.msg.GetPoolType())
+		err = osmoutils.CreateModuleAccount(suite.Ctx, suite.App.AccountKeeper, pool.GetAddress())
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		err = bankKeeper.SendCoins(suite.Ctx, sender, pool.GetAddress(), initialPoolLiquidity)
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		swapModule, err := swaprouterKeeper.GetPoolModule(suite.Ctx, poolId)
+		suite.Require().NoError(err, "test: %v", test.name)
+
+		if test.usingMock {
+			ctrl := gomock.NewController(suite.T())
+			mockPool := mocks.NewMockConcentratedPoolExtension(ctrl)
+			mockPool.EXPECT().GetTotalShares().Return(pool.GetTotalShares()).AnyTimes()
+			mockPool.EXPECT().GetId().Return(poolId).AnyTimes()
+			err = swapModule.InitializePool(suite.Ctx, pool, sender)
+		} else {
+			err = swapModule.InitializePool(suite.Ctx, pool, sender)
+		}
+		// attempt to create a pool with the given NewMsgCreateBalancerPool message. After that,
+		// this function calls the InitializePool function
+		// poolId, err := swaprouterKeeper.CreatePool(suite.Ctx, test.msg)
+
+		if test.expectPass {
+			suite.Require().NoError(err, "test: %v", test.name)
+
+			// check to make sure new pool exists and has minted the correct number of pool shares
+			pool, err := gammKeeper.GetPoolAndPoke(suite.Ctx, poolId)
+			suite.Require().NoError(err, "test: %v", test.name)
+			suite.Require().Equal(types.InitPoolSharesSupply.String(), pool.GetTotalShares().String(),
+				fmt.Sprintf("share token should be minted as %s initially", types.InitPoolSharesSupply.String()),
+			)
+
+			// get expected tokens in new pool and corresponding pool shares
+			expectedPoolTokens := sdk.Coins{}
+			for _, asset := range test.msg.InitialLiquidity() {
+				expectedPoolTokens = expectedPoolTokens.Add(asset)
+			}
+			expectedPoolShares := sdk.NewCoin(types.GetPoolShareDenom(pool.GetId()), types.InitPoolSharesSupply)
+
```

Comment:
should we also check if the expected Pool share and the expected pool token matches the actual tokens and shares of / in the pool?

Reply:
I can do that!